### PR TITLE
added get article by id model, controller, route, + tests

### PR DESCRIPTION
**Pull request by andreasdk**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 const request = require('supertest');
+const express = require('express');
 const app = require('../app');
 const seed = require('../db/seeds/seed')
 const testData = require('../db/data/test-data/index')
 const db = require('../db/connection')
 
+app.use(express.json());
+
```
Comment (daviidbartlett) [inline __tests__/app.test.js:9]: is this needed?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -26,4 +29,36 @@ describe('/api/topics', () => {
...
+                    })
+                    expect(response.body.article).toEqual({
+                        article_id: 1,
+                        title: 'Living in the shadow of a great man',
+                        topic: 'mitch',
+                        author: 'butter_bridge',
+                        body: 'I find this existence challenging',
+                        created_at: '2020-07-09T20:11:00.000Z',
+                        votes: 100
+                    })
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: brittle testing, you may be adding an extra key at some point in which this test will fail 👀

File: __tests__/app.test.js [hunk 3]
```diff
@@ -26,4 +29,36 @@ describe('/api/topics', () => {
...
+        
+            })
+        })
+        test('status: 404 - responds with a 404 if route does not exist.', () => {
+            return request(app).get('/api/artikles/1').expect(404).then(({ body }) => {
+                expect(body.msg).toBe('Error 404 - Route not found');
+            })
+        })
```
Comment (daviidbartlett) [inline __tests__/app.test.js:59]: you don't need to test this for every endpoint!
 you will have a 404 on this endpoint though, not for the path not found

**Review by daviidbartlett — COMMENTED** [review]
few comments, missing error handling

File: __tests__/app.test.js (hunk 3 above)
Comment (jimstev2447) [reply __tests__/app.test.js:59]: still missing a 404 test here 👀

**Review by Augs0 — CHANGES_REQUESTED** [review]
I'd suggest you break your error testing out of your endpoint block for clarity (these should be global). Also, you're using the PSQL code 23503 for PSQL errors to handle 404. This code is actually for foreign key violations (see here: https://www.postgresql.org/docs/8.2/errcodes-appendix.html). Should this be being used for paths not found?

**Pull request merged**